### PR TITLE
Avoid perform trainer.train() for seed checkpoint creation case

### PR DESCRIPTION
**Pull request by fegin**
This pull request splits the `Trainer.__init__()` method to separate necessary initializations from actions that require an initialized trainer. The main() function now handles these actions, which currently include seed checkpoint creation and training step.

This modification resolves the current seed checkpoint failure and enables the implementation of future actions that are not part of the training process.

### CODE REVIEW

File: torchtitan/train.py [hunk 1]
```diff
@@ -460,7 +449,19 @@ def close(self) -> None:
...
+            assert int(
+                os.environ["WORLD_SIZE"]
+            ), "Must create seed checkpoint using a single device, to disable sharding."
+            assert (
+                config.checkpoint.enable_checkpoint
+            ), "Must enable checkpointing when creating a seed checkpoint."
+            trainer.checkpointer.save(curr_step=0, force=True)
+            logger.info("Created seed checkpoint")
+        else:
+            trainer.checkpointer.load(step=config.checkpoint.load_step)
```
Comment (tianyu-l) [inline torchtitan/train.py]: I slightly prefer we still put `load` in `train()`, for the following reasons:
1. the save logic is still in `train()`
2. it'd be more straightforward to understand what makes `self.step` change in `logger.info("Training starts at step {self.step + 1}.")`
3. user should be able to write minimal code to reuse components in `train.py`. In some case, they'll inherit and modify `Trainer` and still directly call `trainer.train()`. Putting this line in `train.py` avoids explicitly calling it for those use cases.

**Review by tianyu-l — APPROVED** [review]
had a nit comment

File: torchtitan/train.py (hunk 1 above)
Comment (fegin) [reply torchtitan/train.py]: make sense. Let's put it into train().
